### PR TITLE
feat: add Map and Set support

### PR DESCRIPTION
**Pull request by gemwuu**

### CODE REVIEW

**Review by fengmk2 — APPROVED** [review]
+1

File: lib/index.js [hunk 1]
```diff
@@ -10,10 +10,11 @@ const cache = new Map();
  *
  * @param {Object} obj
  * @param {string} key
- * @param {!Function|Object} value
+ * @param {!Function|Object|Map|Set|String} value
```
Comment (fent) [inline lib/index.js]: this can be changed to just `Object` as that covers everything

**Review by fent — CHANGES_REQUESTED** [review]
ah interesting, nice catch!

**Review by fent — CHANGES_REQUESTED** [review]
i don't think it's necessary to include examples in the readme about mocking Map/Set. these are edge cases that i wouldn't think many users would find useful when they're trying to learn how to use the library

**Comment by gemwuu** [comment]
> i don't think it's necessary to include examples in the readme about mocking Map/Set. these are edge cases that i wouldn't think many users would find useful when they're trying to learn how to use the library

fair enough, the `README.md` modification has been reverted.

**Review by fent — APPROVED** [review]

**Comment by fent** [comment]
thanks again! @popomore can you create a patch release for this? if not, i can do it late sunday/monday as i'm away from my computer right now

**Comment by popomore** [comment]
@fent I have released 3.0.1

**Pull request merged**